coffee_github/src/repository.rs

```diff
@@ -98,18 +78,42 @@ impl Github {
 
                             let conf_file = serde_yaml::from_str::<Conf>(&conf_str).unwrap();
                             let conf_lang = (&conf_file.plugin.lang).to_owned();
-                            plugin_lang = match conf_lang.as_str() {
-                                "py" => PluginLang::Python,
-                                "go" => PluginLang::Go,
-                                "rs" => PluginLang::Rust,
-                                "dart" => PluginLang::Dart,
-                                "js" => PluginLang::JavaScript,
-                                "ts" => PluginLang::TypeScript,
-                                _ => continue,
+                            match conf_lang.as_str() {
+                                "py" => plugin_lang = PluginLang::Python,
+                                "go" => plugin_lang = PluginLang::Go,
+                                "rs" => plugin_lang = PluginLang::Rust,
+                                "dart" => plugin_lang = PluginLang::Dart,
+                                "js" => plugin_lang = PluginLang::JavaScript,
+                                "ts" => plugin_lang = PluginLang::TypeScript,
+                                _ => return Err(CoffeeError::new(1, &format!("language {conf_lang} not supported"))),
                             };
                             conf = Some(conf_file)
                         }
                     }
+                    
+                    // check if there was a coffee configuration file 
+                    if conf == None
+                    {
+                        // try to understand the language from the file
+                        let files = WalkDir::new(plugin_path.path()).max_depth(1);
+                        for file in files {
+                            let file_dir = file.unwrap().clone();
+                            (path_to_plugin, plugin_name) =
+                                get_plugin_info_from_path(file_dir.path()).unwrap();
+
+                            let file_name = file_dir.file_name().to_str().unwrap();
+                            plugin_lang = match file_name {
+                                "requirements.txt" => PluginLang::Python,
+                                "go.mod" => PluginLang::Go,
+                                "cargo.toml" => PluginLang::Rust,
+                                "pubspec.yaml" => PluginLang::Dart,
+                                "package.json" => PluginLang::JavaScript,
+                                "tsconfig.json" => PluginLang::TypeScript,
+                                _ => continue,
```

Comment:
```suggestion
                                _ => PluginLang::Unknown,
```

Should be available this, right?